ethereum/p2p/src/test/java/tech/pegasys/pantheon/ethereum/p2p/network/DefaultP2PNetworkTest.java

```diff
@@ -122,13 +124,29 @@ public void addingRepeatMaintainedPeersReturnsFalse() {
   @Test
   public void checkMaintainedConnectionPeersTriesToConnect() {
     final DefaultP2PNetwork network = mockNetwork();
+    network.start();
+
     final Peer peer = mockPeer();
     network.peerMaintainConnectionList.add(peer);
 
     network.checkMaintainedConnectionPeers();
     verify(network, times(1)).connect(peer);
   }
 
+  @Test
+  public void checkMaintainedConnectionPeersDoesNotConnectToDisallowedPeer() {
```

Comment:
@lucassaldanha can you confirm that the permissioning controls should prevent peers added via staticnodes.json or admin_addPeer from connecting?  It makes sense to  me but I seem to recall some discussions and details...

Reply:
For the staticnodes.json list we apply the same rules as for bootnodes: they must be permitted.

For admin_addPeer is the same logic as connecting to another peer. If the peer is not allowed it won't connect to the peer (and I believe the addPeer method will give you a nice error response saying why).

Reply:
The "admin_addPeer" method will no longer return error messages if a non-permitted peer is added.  Discussed this with @lucassaldanha offline and we determined that there is no need to keep these error messages.